File: py2many/list_typed.go
package main

import (
	iter "github.com/hgfischer/go-iter"
)

func FillInList(size int) None {
	var l List = []None{}
	for _, i := range iter.NewIntSeq(iter.Start(0), iter.Stop(size)).All() {
		l = append(l, (i + 1))
	}
	return None(l)
}

File: py2many/list_typed.cpp
#include <cppitertools/range.hpp> // NOLINT(build/include_order)
#include <vector>                 // NOLINT(build/include_order)
inline auto fill_in_list(int size) {
  std::vector<
      decltype(std::declval<typename decltype(range(size))::value_type>() + 1)>
      l = {};
  for (auto i : iter::range(size)) {
    l.push_back(i + 1);
  }
  return l;
}
